storage/memory/memory_test.go

```diff
@@ -281,10 +301,48 @@ func TestObjects(t *testing.T) {
 		}
 	}
 	if cnt != 3 {
-		t.Errorf("g.Objects(%s, %s) failed to retrieve 3 objects, got %d instead", ts[0].Subject(), ts[0].Predicate(), cnt)
+		t.Errorf("g.Objects(%s, %s) failed to retrieve 3 objects, got %d instead ", ts[0].Subject(), ts[0].Predicate(), cnt)
 	}
 }
 
+func TestObjectsOffset(t *testing.T) {
+	ts, ctx := getTestOffsetTriples(t), context.Background()
+	g, _ := NewStore().NewGraph(ctx, "test")
+	if err := g.AddTriples(ctx, ts); err != nil {
+		t.Errorf("g.AddTriples(_) failed failed to add test triples with error %v", err)
+	}
+	// To avoid blocking on the test. On a real usage of the driver you would like
+	// to call the graph operation on a separated goroutine using a sync.WaitGroup
+	// to collect the error code eventually.
+	var objs, objs1, objs2 []string
+	objStr := make(map[string]string)
+	for _, ob := range ts {
+		objStr[ob.String()] = ob.Object().String()
+		objs = append(objs, ob.String())
+	}
+	sort.Strings(objs)
+	for _, ob := range objs {
+		objs1 = append(objs1, objStr[ob])
+	}
+
+	for i, ob := range ts {
+		os := make(chan *triple.Object, 1)
+		lo := &storage.LookupOptions{MaxElements: 1, Offset: i % 3}
+		if err := g.Objects(ctx, ob.Subject(), ob.Predicate(), lo, os); err != nil {
+			t.Errorf("g.Objects(%s, %s) failed with error %v", ob.Subject(), ob.Predicate(), err)
+		}
+		a := <-os
+		objs2 = append(objs2, a.String())
+	}
+
+	for i := range objs1 {
```

Comment:
Prefer using cmp.Equal or cmp.Diff instead of doing this comparison manually.

cmp.Equal: https://pkg.go.dev/github.com/google/go-cmp/cmp#Equal
cmp.Diff: https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff

Reply:
To use the cmp library is necessary to add a dependency so we discard using it